exonum/src/node/consensus.rs

```diff
@@ -141,137 +141,111 @@ impl NodeHandler {
 
         if has_unknown_txs {
             trace!("REQUEST TRANSACTIONS");
-            self.request(RequestData::Transactions(hash), from);
+            self.request(RequestData::ProposeTransactions(hash), from);
 
             for node in known_nodes {
-                self.request(RequestData::Transactions(hash), node);
+                self.request(RequestData::ProposeTransactions(hash), node);
             }
         } else {
             self.handle_full_propose(hash, msg.round());
         }
     }
 
-    // Validates transaction from the block and appends them into the pool.
-    // Returns tuple of transaction hashes and Patch of changes in the pool.
-    fn validate_block_transactions(&self, block: &BlockResponse) -> Option<(Vec<Hash>, Patch)> {
-        let mut fork = self.blockchain.fork();
-        let mut tx_hashes = Vec::new();
-        {
-            let mut schema = Schema::new(&mut fork);
-            for raw in block.transactions() {
-                let tx = match self.blockchain.tx_from_raw(raw) {
-                    Ok(tx) => tx,
-                    Err(e) => {
-                        error!("{}, block={:?}", e.description(), block);
-                        return None;
-                    }
-                };
-
-                let hash = tx.hash();
-                if schema.transactions().contains(&hash)
-                    && !schema.transactions_pool().contains(&hash)
-                {
-                    error!(
-                        "Received block with already committed transaction, block={:?}",
-                        block
-                    );
-                    return None;
-                }
-                profiler_span!("tx.verify()", {
-                    if !tx.verify() {
-                        error!("Incorrect transaction in block detected, block={:?}", block);
-                        return None;
-                    }
-                });
-                schema.add_transaction_into_pool(tx.raw().clone());
-                tx_hashes.push(hash);
-            }
-        }
-        Some((tx_hashes, fork.into_patch()))
-    }
-
-    /// Handles the `Block` message. For details see the message documentation.
-    // TODO write helper function which returns Result (ECR-123)
-    #[cfg_attr(feature = "flame_profile", flame)]
-    pub fn handle_block(&mut self, msg: &BlockResponse) {
-        // Request are sent to us
+    fn validate_block_response(&self, msg: &BlockResponse) -> Result<(), String> {
+        // Request are sent to us.
```

Comment:
Either `Request is` or `Requests are`.
And I suggest to make comment more verbose, e.g. `Check that request is sent to us` (or remove it completely, the next line is pretty self-explanatory).